src/test/regress/expected/distributed_functions.out

```diff
@@ -829,6 +829,563 @@ SELECT * FROM test ORDER BY id;
 (2 rows)
 
 DROP TABLE test;
+-- verify that recreating distributed functions with TABLE params gets propagated to workers
+CREATE OR REPLACE FUNCTION func_with_return_table(int)
+RETURNS  TABLE (date date)
+LANGUAGE plpgsql AS $$
+BEGIN
+    RETURN query SELECT '2011-01-01'::date;
+END;
+$$;
+SELECT create_distributed_function('func_with_return_table(int)');
+ create_distributed_function
+---------------------------------------------------------------------
+
+(1 row)
+
+SET citus.log_remote_commands TO ON;
+CREATE OR REPLACE FUNCTION func_with_return_table(int)
+RETURNS  TABLE (date date)
+LANGUAGE plpgsql AS $$
+BEGIN
+    RETURN query SELECT '2011-01-01'::date;
+END;
+$$;
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx, xx, 'xxxxxxx');
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx, xx, 'xxxxxxx');
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing SET citus.enable_ddl_propagation TO 'off'
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing SET citus.enable_ddl_propagation TO 'off'
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing CREATE OR REPLACE FUNCTION function_tests.func_with_return_table(integer)
+ RETURNS TABLE(date date)
+ LANGUAGE plpgsql
+AS $function$
+BEGIN
+    RETURN query SELECT '2011-01-01'::date;
+END;
+$function$
+
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing CREATE OR REPLACE FUNCTION function_tests.func_with_return_table(integer)
+ RETURNS TABLE(date date)
+ LANGUAGE plpgsql
+AS $function$
+BEGIN
+    RETURN query SELECT '2011-01-01'::date;
+END;
+$function$
+
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
```

Comment:
can we query `pg_proc` on the workers to check if everything matches with the coordinator?

Such as where these types of queries should always return 1 row per given function / pg_proc fields. Below, I used two fields, but use as many as possible:
```SQL
SELECT count(*) FROM (SELECT result FROM  run_command_on_workers($$select row(pg_proc.pronargs, pg_proc.proargtypes) from pg_proc where proname = 'func_with_return_table';$$) UNION  select row(pg_proc.pronargs, pg_proc.proargtypes)::text from pg_proc where proname = 'func_with_return_table') as foo;
 count 
-------
     1
(1 row)
```